### PR TITLE
test: use resource disposal to ensure servers shutdown even if tests fail

### PR DESCRIPTION
**Pull request by nektro**
`expect()` throws if given a non-passing value so many tests forget to cleanup or are wrapped in `try`/`finally`. with `using` they don't have to.

### CODE REVIEW

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
The tests which start a server inside a `beforeAll` / `afterAll` need to continue doing that

It's really easy to use all the CI machine's open ports and then cause other tests to fail

**Comment by nektro** [comment]
doing it this way still only creates one server per test but also has the added benefit that if/once we run tests in parallel they wont clobber the same global variable

File: test/js/web/fetch/fetch.test.ts [hunk 1]
```diff
@@ -1676,25 +1665,9 @@ it("same-origin status code 302 should not strip headers", async () => {
   expect(headers.get("Authorization")).toEqual("yes");
   expect(url).toEndWith("/redirected");
   expect(redirected).toBe(true);
-  server.stop(true);
 });
 
 describe("should handle relative location in the redirect, issue#5635", () => {
-  var server: Server;
```
Comment (Jarred-Sumner) [inline test/js/web/fetch/fetch.test.ts:1683]: can we bring this back

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**